driver.go

```diff
@@ -49,8 +66,16 @@ type Pin interface {
 	Name() string
 	Number() int
 }
+
 type Driver interface {
 	io.Closer
 	Metadata() Metadata
 	Pins(Capability) ([]Pin, error)
 }
+
+type DriverFactory interface {
+	GetParameters() []ConfigParameter
+	ValidateParameters(parameters []ConfigParameter) (bool, []string)
+	Metadata() Metadata
+	CreateDriver(parameters []ConfigParameter, hardwareResources interface{}) (interface{}, error)
```

Comment:
I'm not wild about the interface{} return type, but I'm not sure of any other reasonable alternatives (other than the current HalAdapter implementation by coincidence).

hardwareResources should represent singleton hardware resources like i2c bus, spi bus, and possibly other resources.